RelationalAI/CommonPolicies.cs

```diff
@@ -41,41 +41,47 @@ static CommonPolicies()
         /// </summary>
         /// <typeparam name="T">Result type of an operation.</typeparam>
         /// <param name="policyBuilder">The policy builder.</param>
+        /// <param name="startTime">transaction startTime epoch milliseconds</param>
+        /// <param name="overheadRate">the overhead % to add through polling</param>
         /// <param name="delayThreshold">Max delay between retry attempts in seconds. Defaults to 30 seconds.</param>
         /// <returns>Resulting policy instance.</returns>
-        public static AsyncPolicy<T> RetryForeverWithBoundedDelay<T>(this PolicyBuilder<T> policyBuilder, int delayThreshold = 30)
+        public static AsyncPolicy<T> RetryForeverWithBoundedDelay<T>(this PolicyBuilder<T> policyBuilder, long startTime, double overheadRate = 0.01, int delayThreshold = 120)
         {
             return policyBuilder
-                .WaitAndRetryForeverAsync(retryAttempt => GetBoundedRetryDelay(retryAttempt, delayThreshold))
+                .WaitAndRetryForeverAsync(retryAttempt => GetBoundedRetryDelay(retryAttempt, startTime, overheadRate, delayThreshold))
                 .AddRequestErrorResilience();
         }
 
         /// <summary>
         /// Creates a policy that can be used to produce synchronous API for async calls which may be taking less than 15 min
         /// to complete, otherwise throws TimeoutRejectedException.
-        /// Uses exponential back-off starting with a 2 seconds delay up to 15 seconds and then retrying every 15 seconds.
+        /// Uses exponential back-off with an overhead % of the time the transaction has been running so far up to 15 seconds and then retrying every 15 seconds.
         /// Retries on HTTP request sending errors.
         /// </summary>
         /// <typeparam name="T">Result type of an operation.</typeparam>
         /// <param name="policyBuilder">The policy builder.</param>
+        /// <param name="startTime">transaction startTime epoch milliseconds</param>
+        /// <param name="overheadRate">the overhead % to add through polling</param>
         /// <returns>Resulting policy instance.</returns>
-        public static AsyncPolicy<T> Retry15Min<T>(this PolicyBuilder<T> policyBuilder)
+        public static AsyncPolicy<T> Retry15Min<T>(this PolicyBuilder<T> policyBuilder, long startTime, double overheadRate = 0.01)
```

Comment:
do we think there is real value introducing 2 methods with hardcoded timeout values in their names (`Retry15Min` & `Retry30Min`)?

Reply:
I don't but there were introduced at some point, I think it's not harmful to keep them but we can remove those methods if required